### PR TITLE
Update docs with TF mixed-precision optimization

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

**Comment by deadeyegoodwin** [comment]
Related to #1472

File: docs/optimization.rst [hunk 1]
```diff
@@ -308,6 +306,27 @@ cutting latency in half. The benefit provided by TensorRT will vary
 based on the model, but in general it can provide significant
 performance improvement.
 
+TensorFlow Automatic FP16 Optimization
+......................................
+
+TensorFlow has another option to provide FP16 optimization that can be
```
Comment (CoderHam) [inline docs/optimization.rst:312]: provide speedup by using the FP16

Comment (deadeyegoodwin) [reply docs/optimization.rst:312]: I don't want to say "speedup" explicitly. Already calling this an optimization.

Comment (CoderHam) [reply docs/optimization.rst:312]: Makes sense. Resolving

**Pull request merged**